File: tests/fixtures/sample_ts/calculator.ts
export class Calculator {
  private value: number;

  constructor(initial: number = 0) {
    this.value = initial;
  }

  add(n: number): Calculator {
    this.value += n;
    return this;
  }

  subtract(n: number): Calculator {
    this.value -= n;
    return this;
  }

  multiply(n: number): Calculator {
    this.value *= n;
    return this;
  }

  getResult(): number {
    return this.value;
  }

  reset(): void {
    this.value = 0;
  }
}

export function createCalculator(initial?: number): Calculator {
  return new Calculator(initial);
}

export interface CalculatorOptions {
  initialValue?: number;
  precision?: number;
}

File: tests/fixtures/sample_rb/calculator.rb
module Calculator
  class Engine
    attr_reader :value, :name

    def initialize(initial = 0, name = "default")
      @value = initial
      @name = name
    end

    def add(n)
      @value += n
      puts "Adding #{n}, result: #{@value}"
      self
    end

    def subtract(n)
      @value -= n
      self
    end

    def multiply(n)
      @value *= n
      self
    end

    def result
      @value
    end

    def reset
      @value = 0
      self
    end
  end

  class AdvancedEngine < Engine
    attr_reader :history

    def initialize(initial = 0, name = "advanced")
      super(initial, name)
      @history = []
    end

    def add(n)
      @history << { operation: :add, value: n, result: @value + n }
      super(n)
    end

    def subtract(n)
      @history << { operation: :subtract, value: n, result: @value - n }
      super(n)
    end

    def last_operation
      @history.last
    end
  end

  module Helpers
    def self.format_result(value)
      "Result: #{value}"
    end

    def self.validate_input(input)
      raise ArgumentError, "Input must be numeric" unless input.is_a?(Numeric)
      true
    end
  end
end
